lib/omniauth/login_dot_gov/logout_utility.rb

```diff
@@ -0,0 +1,60 @@
+module OmniAuth
+  module LoginDotGov
+    # Assists in RP-initiated logout: https://developers.login.gov/oidc/#logout
+    # @example RP-initiated logout in Rails controller
+    #   class SessionsController < Devise::SessionsController
+    #     def destroy
+    #       logout_request = self.class.logout_utility.build_request(id_token: session[:id_token],
+    #         post_logout_redirect_uri: 'http://localhost:3001/'
+    #       )
+    #       sign_out(current_user)
+    #       redirect_to(logout_request.redirect_uri) and return
+    #     end
+    #
+    #     # Avoid making multiple HTTP requests by memoizing utility class
+    #     def self.logout_utility
+    #       @logout_utility ||=
+    #         OmniAuth::LoginDotGov::LogoutUtility.new(idp_base_url: Rails.configuration.oidc['idp_url'])
+    #     end
+    #   end
+    class LogoutUtility
+      attr_reader :state
+
+      Request = Struct.new(:redirect_uri, :state)
+
+      # Initializes with one of idp_base_url, idp_configuration or end_session_endpoint.
+      # Initializing with idp_base_url and idp_configuration may send an HTTP request to
+      # fetch the OpenID configuration. The object should be memoized to avoid sending an
+      # HTTP request for each logout.
+      def initialize(idp_base_url: nil, idp_configuration: nil, end_session_endpoint: nil)
+        unless idp_base_url || end_session_endpoint || idp_configuration
+          raise ArgumentError('idp_base_url, end_session_endpoint or idp_configuration must not be nil')
+        end
```

Comment:
Looks like CC got mad at the cognitive complexity..... an idea to see if this will appease

```suggestion
      def check_initialize_arguments!(idp_base_url, end_session_endpoint, idp_configuration)
        unless idp_base_url || end_session_endpoint || idp_configuration
          raise ArgumentError('idp_base_url, end_session_endpoint or idp_configuration must not be nil')
        end
      end

      def initialize(idp_base_url: nil, idp_configuration: nil, end_session_endpoint: nil)
        check_initialize_arguments!(idp_base_url, end_session_endpoint, idp_configuration)
```